src/main/java/com/assignment1/parser/Validator.java

```diff
@@ -0,0 +1,157 @@
+   * It should be valid and of right format to be correct.
+   */
+  public boolean checkOption(final String option,
+                             final String currentString,
+                             final String nextString,
+                             final boolean alreadyOccurred) throws RuntimeExceptionCustom {
+    boolean result = false;
+    if (isValidOption(option, currentString, nextString, alreadyOccurred)) {
+      if (checkFormat(option, nextString)) {
+        result = true;
+      } else {
+        throw customException;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Explicit check for name option because it contains multiple words.
+   * Return count of words or throws exception if name not present blank.
+   */
+  public int checkName(final String... args) throws RuntimeExceptionCustom {
+    int numWords = 0;
+    if ("-name".equals(args[0])) { //First argument mandatory to be name
+      numWords += 1; //count this as significant argument
+      int ptr = 1; //pointer to check for how many words name is extended to.
+      while (ptr < args.length && args[ptr].charAt(0) != '-') {
+        //loop until we get option with first character '-'
+        if (!checkFormat("-name", args[ptr])) {
+          //If Name format wrong, then throw Exception
+          throw customException;
+        }
+        numWords += 1; //Increase count for significant arguments
+        ptr++;
+      }
+    } else {
+      throw customException;
+    }
+    return numWords;
+  }
+
+  /**
+   * Implementation for Validate method.
+   */
+  @Override
+  public ItemEntity validate(final String[] args) throws RuntimeExceptionCustom {
+    final int len = args.length;
+    int countValidArgs = 0;
+    final ItemEntity item = new ItemEntity();
+
+    countValidArgs += checkName(args); //Either exception will be thrown or integer returned
+
+    final StringBuilder stringBuilder = new StringBuilder();
+    for (int i = 1; i < countValidArgs - 1; i++) {  //Building up the full name of item
+      stringBuilder.append(args[i]).append(" ");
+    }
+    stringBuilder.append(args[countValidArgs - 1]); //Putting last word separately to avoid space
+    item.setName(stringBuilder.toString());
+
+    boolean typePresent = false;
+    boolean quantityPresent = false;
+    boolean pricePresent = false;
+
+    for (int i = 0; i < len - 1; i++) { //loop to find other options and values
+      if (checkOption("-price", args[i], args[i + 1], pricePresent)) {
+        countValidArgs += 2; //Count increase by 2 for option and value
+        pricePresent = true;
+        item.setPrice(Double.parseDouble(args[i + 1]));
+      }
+
+      if (checkOption("-quantity", args[i], args[i + 1], quantityPresent)) {
+        countValidArgs += 2;
+        quantityPresent = true;
+        item.setQuantity(Integer.parseInt(args[i + 1]));
+      }
+
+      if (checkOption("-type", args[i], args[i + 1], typePresent)) {
+        countValidArgs += 2;
```

Comment:
this whole function needs to be simplified

Reply:
Simplified a little bit. Also, moved this logic to parser instead of validator.